sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -402,9 +403,49 @@ export class StreamingReceiver extends MessageReceiver {
     await this._receiverHelper.suspend();
   }
 
-  async init(useNewName: boolean, abortSignal?: AbortSignalLike): Promise<void> {
-    const options = this._createReceiverOptions(useNewName, this._getHandlers());
-    await this._init(options, abortSignal);
+  async init(
+    // TODO: make onError? required
+    args: { useNewName: boolean; connectionId: string; onError?: OnError } & Pick<
+      OperationOptionsBase,
+      "abortSignal"
+    >
+  ) {
+    const config: RetryConfig<void> = {
+      operation: async () => {
+        try {
+          return this._initOnce(args);
+        } catch (err) {
+          args.onError &&
+            args.onError({
+              error: err,
+              errorSource: "receive",
+              entityPath: this.entityPath,
+              fullyQualifiedNamespace: this._context.config.host
+            });
+
+          // once we're at this point where we can spin up a connection we're past the point
+          // of fatal errors like the connection string just outright being malformed, for instance.
+          err.retryable = true;
```

Comment:
I dont see us rethrowing the error, so how does the retry() method know to keep trying?

Reply:
Yes, I need to add actual testing. This concept seems okay to you - it seems "right" in that it reflects what we're saying - this particular receiver method is "special" so we'll still translate but we have our own policy for the retryable flag.

Reply:
Yes, sounds good. Check with @JoshLove-msft if they change the `IsTransient` flag on the error when calling user's error handler.

Reply:
Additional thought as I add tests - this does not apply for abortSignals. :)

Reply:
I rearranged this a bit. The error does not have the retryable flag modified when it gets reported to processError. We only modify the retryable flag just before throwing it.

So this change is now no longer user-visible.